diff --git a/cjs/index.js b/cjs/index.js
index 740f76e0efe40f874865a3d85b5779085f0b8803..b48ce3a044a624a9c09e9f5cd06e7d9fb62a9cbc 100644
--- a/cjs/index.js
+++ b/cjs/index.js
@@ -185,7 +185,7 @@ exports.preset = {
                 schema: options.schema,
                 config: {
                     inlineFragmentTypes: isMaskingFragments ? 'mask' : options.config['inlineFragmentTypes'],
-                    ...forwardedConfig,
+                    ...options.config,
                 },
                 documents: sources,
                 documentTransforms: options.documentTransforms,
diff --git a/esm/index.js b/esm/index.js
index 0c57c88f943890833e9f1bb203316cce04ffb470..1cb88057e43a2267f57b678d7b33b8760ee3d5e9 100644
--- a/esm/index.js
+++ b/esm/index.js
@@ -180,7 +180,7 @@ export const preset = {
                 schema: options.schema,
                 config: {
                     inlineFragmentTypes: isMaskingFragments ? 'mask' : options.config['inlineFragmentTypes'],
-                    ...forwardedConfig,
+                    ...options.config,
                 },
                 documents: sources,
                 documentTransforms: options.documentTransforms,
