diff --git a/cjs/index.js b/cjs/index.js
index 69bc3334efbb29ac1ddef98304a8fa09ffcdef8e..a2521f5fb92152710d8366dbcd8f6726e699ef2d 100644
--- a/cjs/index.js
+++ b/cjs/index.js
@@ -190,7 +190,7 @@ exports.preset = {
                 schema: options.schema,
                 config: {
                     inlineFragmentTypes: isMaskingFragments ? 'mask' : options.config['inlineFragmentTypes'],
-                    ...forwardedConfig,
+                    ...options.config,
                 },
                 documents: sources,
                 documentTransforms: options.documentTransforms,
diff --git a/esm/index.js b/esm/index.js
index 9147ef68449454dc144b0663c9a5af33c95257b3..2c3247ae9df163ff00354fd2d89b1a0bdb27d03b 100644
--- a/esm/index.js
+++ b/esm/index.js
@@ -7,6 +7,7 @@ import { ClientSideBaseVisitor, DocumentMode } from '@graphql-codegen/visitor-pl
 import * as fragmentMaskingPlugin from './fragment-masking-plugin.js';
 import { generateDocumentHash, normalizeAndPrintDocumentNode } from './persisted-documents.js';
 import { processSources } from './process-sources.js';
+export { addTypenameSelectionDocumentTransform } from './add-typename-selection-document-transform.js';
 export { default as babelOptimizerPlugin } from './babel.js';
 const isOutputFolderLike = (baseOutputDir) => baseOutputDir.endsWith('/');
 export const preset = {
@@ -185,7 +186,7 @@ export const preset = {
                 schema: options.schema,
                 config: {
                     inlineFragmentTypes: isMaskingFragments ? 'mask' : options.config['inlineFragmentTypes'],
-                    ...forwardedConfig,
+                    ...options.config,
                 },
                 documents: sources,
                 documentTransforms: options.documentTransforms,
@@ -241,4 +242,4 @@ function createDeferred() {
     });
     return d;
 }
-export { addTypenameSelectionDocumentTransform } from './add-typename-selection-document-transform.js';
+
